lib/figaro_yaml_validator.rb

```diff
@@ -0,0 +1,30 @@
+class FigaroYamlValidator
+  FILE_PATH = File.join(Rails.root, 'config', 'application.yml')
+  FIGARO_YAML = YAML.load(IO.read(FILE_PATH))
+
+  def validate(yaml_data = FIGARO_YAML)
+    bad_keys = []
+    check_for_bad_values(yaml_data, bad_keys)
+    return unless bad_keys.any?
+    raise warning(bad_keys).tr("\\\n", ' ')
+  end
+
+  private
+
+  def check_for_bad_values(yaml_data, bad_keys)
+    yaml_data.map do |key, value|
+      if value.is_a?(Hash)
+        check_for_bad_values(value, bad_keys)
+      elsif value == 'yes' || value == 'no'
+        bad_keys << key
+      end
+    end
+  end
+
+  def warning(bad_keys)
+    <<~HEREDOC
+      You have invalid values for #{bad_keys.to_sentence} in
+      config/application.yml. Please change them to true or false
+    HEREDOC
+  end
+end
```

Comment:
I like the idea of checking the whole file rather than just the env-specific stuff because in ensures we aren't going to use invalid values in any env. Does that make sense?

I see what you are saying about the error message not being specific to env, but I think someone should pretty easily be able to grep and find the right key?

Because I am checking all envs, I also need to deal with a nested hash, which is why this is more complicated.